cli/src/main/scala/com/lightbend/rp/reactivecli/runtime/kubernetes/Deployment.scala

```diff
@@ -145,53 +148,62 @@ object Deployment {
-              "replicas" -> noOfReplicas.asJson,
-              "serviceName" -> appVersionMajor.asJson,
-              "template" -> Json(
-                "app" -> appName.asJson,
-                "appVersionMajor" -> appVersionMajor.asJson,
-                "appVersionMajorMinor" -> appVersionMajorMinor.asJson,
-                "appVersion" -> appVersion.asJson),
+          Deployment(
+            appVersion,
+            Json(
+              "apiVersion" -> apiVersion(kubernetesVersion).asJson,
+              "kind" -> "Deployment".asJson,
+              "metadata" -> Json(
+                "name" -> appVersion.asJson,
+                "labels" -> Json(
+                  "app" -> appName.asJson,
+                  "appVersionMajor" -> appVersionMajor.asJson,
+                  "appVersionMajorMinor" -> appVersionMajorMinor.asJson,
+                  "appVersion" -> appVersion.asJson)),
               "spec" -> Json(
-                "containers" -> List(
-                  Json(
-                    "name" -> appName.asJson,
-                    "image" -> imageName.asJson,
-                    "imagePullPolicy" -> imagePullPolicy.asJson,
-                    "env" -> annotations.environmentVariables.asJson,
-                    "ports" -> annotations.endpoints.asJson)
-                    .deepmerge(annotations.readinessCheck.asJson(readinessProbeEncode))
-                    .deepmerge(annotations.healthCheck.asJson(livenessProbeEncode))).asJson))))
+                "replicas" -> noOfReplicas.asJson,
+                "serviceName" -> appVersionMajor.asJson,
+                "template" -> Json(
+                  "app" -> appName.asJson,
+                  "appVersionMajor" -> appVersionMajor.asJson,
+                  "appVersionMajorMinor" -> appVersionMajorMinor.asJson,
+                  "appVersion" -> appVersion.asJson),
+                "spec" -> Json(
+                  "containers" -> List(
+                    Json(
+                      "name" -> appName.asJson,
+                      "image" -> imageName.asJson,
+                      "imagePullPolicy" -> imagePullPolicy.asJson,
+                      "env" -> annotations.environmentVariables.asJson,
+                      "ports" -> annotations.endpoints.asJson)
+                      .deepmerge(annotations.readinessCheck.asJson(readinessProbeEncode))
+                      .deepmerge(annotations.healthCheck.asJson(livenessProbeEncode))).asJson)))))
       case _ =>
         Failure(new IllegalArgumentException("Unable to generate Kubernetes Deployment: both application name and version are required"))
     }
 
-  private[kubernetes] def apiVersion(kubernetesVersion: (KubernetesMajorVersion, KubernetesMinorVersion)): String = {
-    val (major, minor) = kubernetesVersion
-    if (major >= 1 && minor >= 8)
+  private[kubernetes] def apiVersion(kubernetesVersion: KubernetesVersion): String =
+    if (kubernetesVersion.major >= 1 && kubernetesVersion.minor >= 8)
```

Comment:
Won't this break with 2.0 release? Perhaps something like `kubernetesVersion.major && (kubernetesVersion.major > 1 || kubernetesVersion.minor >= 8`?

Reply:
Oops... 😳 You're right - I have to fix this.

This should teach me a thing or two about coding on jetlag induced haze with an empty stomach. I should have had that Panda Express 🐼 🍱  before coding this part.

Reply:
This is now fixed - thanks for catching this!

Reply:
> Panda Express

I haven't had good experiences there :face_with_thermometer: